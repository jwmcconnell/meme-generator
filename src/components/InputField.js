import React from 'react';
import PropTypes from 'prop-types';

const InputField = ({ label, type, update, defaultValue, min }) => {
  return (
    <label>{label}
      <input 
        type={type} 
        onChange={update} 
        defaultValue={defaultValue ? defaultValue : undefined}
        min={min ? min : undefined}
      />
    </label>
  );
};

InputField.propTypes = {
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  update: PropTypes.func.isRequired,
  defaultValue: PropTypes.string,
  min: PropTypes.string
};

export default InputField;
