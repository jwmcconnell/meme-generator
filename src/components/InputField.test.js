/* eslint-disable no-console */
import React from 'react';
import { shallow } from 'enzyme';
import InputField from './InputField';

describe('InputField component', () => {
  it('renders InputField', () => {
    
    const update = e => console.log(e.target.value);
    
    const wrapper = shallow(
      <InputField 
        label='someLabel'
        name='someName'
        type='text'
        update={update}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
  
