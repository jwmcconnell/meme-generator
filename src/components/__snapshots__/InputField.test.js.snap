// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component renders InputField 1`] = `
<label>
  someLabel
  <input
    name="someName"
    onChange={[Function]}
    type="text"
  />
</label>
`;
